Assets/SteamVR_Unity_Toolkit/Scripts/VRTK_HeightAdjustTeleport.cs

```diff
@@ -86,31 +120,55 @@ private void DropToNearestFloor(bool withBlink)
                     currentFloor = rayCollidedWith.transform.gameObject;
                     currentRayDownY = floorY;
 
-                    if (withBlink && !rayCollidedWith.transform.GetComponent<MeshCollider>())
+                    float fallDistance = transform.position.y - floorY;
+                    bool shouldDoGravityFall = useGravityFall && (playerPhysics.IsFalling() || fallDistance > gravityFallHeight);
+
+                    if (withBlink && !shouldDoGravityFall && !rayCollidedWith.transform.GetComponent<MeshCollider>())
                     {
                         Blink(blinkTransitionSpeed);
                     }
 
-                    Vector3 newPosition = new Vector3(transform.position.x, floorY, transform.position.z);
-                    var teleportArgs = new DestinationMarkerEventArgs
+                    if (shouldDoGravityFall)
+                    {
+                        playerPhysics.StartPhysicsFall(Vector3.zero);
+                    }
+                    else // teleport fall
                     {
-                        destinationPosition = newPosition,
-                        distance = rayCollidedWith.distance,
-                        enableTeleport = true,
-                        target = currentFloor.transform
-                    };
-                    OnTeleporting(gameObject, teleportArgs);
-                    SetNewPosition(newPosition, currentFloor.transform);
-                    OnTeleported(gameObject, teleportArgs);
+                        Vector3 newPosition = new Vector3(transform.position.x, floorY, transform.position.z);
+                        var teleportArgs = new DestinationMarkerEventArgs
+                        {
+                            destinationPosition = newPosition,
+                            distance = rayCollidedWith.distance,
+                            enableTeleport = true,
+                            target = currentFloor.transform
+                        };
+                        OnTeleporting(gameObject, teleportArgs);
+                        SetNewPosition(newPosition, currentFloor.transform);
+                        OnTeleported(gameObject, teleportArgs);
+                    }
                 }
             }
         }
 
+        private bool isExternalSystemManipulatingPlaySpace()
```

Comment:
upper case method name `isExternalSystemManipulatingPlaySpace` should be `IsExternalSystemManipulatingPlaySpace`